src/Http/Http.Abstractions/src/HttpResponse.cs

```diff
@@ -127,9 +127,13 @@ public abstract class HttpResponse
         /// Starts the response by calling OnStarting() and making headers unmodifiable.
         /// </summary>
         /// <param name="cancellationToken"></param>
-        /// <remarks>
-        /// If the <see cref="IHttpResponseStartFeature"/> isn't set, StartAsync will default to calling HttpResponse.Body.FlushAsync().
-        /// </remarks>
         public virtual Task StartAsync(CancellationToken cancellationToken = default) { throw new NotImplementedException(); }
+
+        /// <summary>
+        /// Flush any remaining response headers, data, or trailers.
+        /// This may throw if the response is in an invalid state such as a Content-Length mismatch.
+        /// </summary>
+        /// <returns></returns>
+        public virtual Task CompleteAsync() { throw new NotImplementedException(); }
```

Comment:
Shouldn't this take a cancellation token?

Reply:
Fowler shot that down already. PipeWriter.CompleteAsync doesn't take one either.

Reply:
Uhhhh that doesn't seem good.

Reply:
I don't love it either, but the comparison was made to DisposeAsync, and at this point I think it's best to stay consistent with PipeWriter. In the meantime, as an app developer, if you're worried about not being able to cancel a flush made by StreamPipeWriter.CompleteAsync(), you should be flushing the PipeWriter before completing it anyway.